File: hw/elbay/u3/hw17OOPConceptsZoo/OOPConceptsZoo_JavaScript_DidierElbay/OOPConceptsZoo.javascript/model/Pig.js
/**
 * Pig - Represents a pig in the farm. Implements the IMeatAnimal contract.
 * @author Didier Elbay
 */
import { FarmAnimal } from "./FarmAnimal.js";
import { IMeatAnimal } from "../controller/IMeatAnimal.js";
import { Cut } from "./Cut.js";

export class Pig extends FarmAnimal {
    constructor(id, breed, bornOn, weight, idealWeight) {
        super(id, breed, bornOn, weight);
        this.idealWeight = idealWeight;
    }

    isReadyForSlaughter() {
        return this.weight >= this.idealWeight;
    }

    sendToButcher() {
        console.log(`Pig ${this.id} sent to butcher.`);
    }

    // IMeatAnimal contract
    cut() {
        return [
            new Cut(1, "Pork Chop", "Standard Cut",   parseFloat((this.weight * 0.15).toFixed(2))),
            new Cut(2, "Ribs",      "Rib Extraction",  parseFloat((this.weight * 0.10).toFixed(2))),
        ];
    }

    sendToSlaughterHouse(slaughterhouse) {
        console.log(`Pig ${this.id} sent to ${slaughterhouse.description}`);
    }
}

File: hw/alban/u3/hw21AniamlFarm/AnimalFarm.py/model/pig.py
from model.farm_animal import FarmAnimal
from controller.i_meat_animal import IMeatAnimal
from model.cut import Cut

class Pig(FarmAnimal, IMeatAnimal):
    def __init__(self, id_animal, breed, born_on, weight, ideal_weight):
        super().__init__(id_animal, breed, born_on, weight)
        self.ideal_weight = ideal_weight

    def send_to_butcher(self):
        print(f"Pig {self.id} sent to butcher.")

    def cut(self):
        cuts = [
            Cut(1, "Pork Chop", "Standard Cut", 2.5),
            Cut(2, "Ribs", "Rib Extraction", 4.0)
        ]
        return cuts

    def send_to_slaughter_house(self, slaughterhouse):
        print(f"Pig {self.id} sent to {slaughterhouse.description}")